File: JavaNgArch/src/javangarch/domain/domainmodel/ta/TaskItem.java
package javangarch.domain.domainmodel.ta;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

import javangarch.domain.domainmodel.gn.User;
import projectbase.domain.BaseDomainObject;
import projectbase.domain.customcollection.IDomainList;
import projectbase.sharparch.domain.domainmodel.DomainSignature;
import lombok.*;

@Getter@Setter
public class TaskItem extends BaseDomainObject {
	
	public static final String DefaultSort = "this.orderNum";

	@DomainSignature
	private Task task;
	@DomainSignature
	private String brief;
	private User user;
	private String requirement;
	private String record;
	private String keyInfo;
	private Date createDate;
	private Date actionDate;
	private Integer status;
	private int orderNum;

}

File: NetNgArch/src/NetNgArch.Domain/DomainModel/TA/TaskItem.cs
using System;
using NetNgArch.Domain.DomainModel.GN;
using ProjectBase.Domain;
using ProjectBase.Utils;
using SharpArch.Domain.DomainModel;

namespace NetNgArch.Domain.DomainModel.TA
{
    public class TaskItem : BaseDomainObject
    {
        public static readonly SortStruc<TaskItem>[] DefaultSort = new SortStruc<TaskItem>[]
                                                                    {
                                                                        new SortStruc<TaskItem>{
                                                                            OrderByExpression=o => o.OrderNum,
                                                                            OrderByDirection = OrderByDirectionEnum.Asc
                                                                        }
                                                               };
        #region "persistent properties"

		[DomainSignature]
		public virtual Task Task { get; set; }
		[DomainSignature]
		public virtual string Brief { get; set; }
		public virtual User User { get; set; }
		public virtual string Requirement { get; set; }
		public virtual string Record { get; set; }
		public virtual string KeyInfo { get; set; }
		public virtual DateTime CreateDate { get; set; }
		public virtual DateTime? ActionDate { get; set; }
		public virtual Task.StatusEnum Status { get; set; }
		public virtual int OrderNum { get; set; }

     
        #endregion


        #region "methods"

        #endregion

        #region "Enums"

        #endregion
    }

}

